### DISCUSSION TITLE
ANRs and Performance Monitoring for my Android Library

### DISCUSSION THREAD

**Discussion by amrita019** [getsentry/sentry-java#2878] [Q&A] [answered]
Hey team,

I've integrated Sentry for Crash monitoring into my Android Library using Hub. This ensures that the logs from the Main Application don't clutter up the system. However, I'm facing an issue where I'm unable to log ANRs or enable Performance Monitoring.

I've looked for documentation on how to maintain two instances of Sentry within an application but couldn't find any. Can you please help me with this?

Thank you.

**Comment by romtsn** [comment, marked as answer]
hi @amrita019, officially we do not support the usecase of using our SDKs in another third-party library/SDK, see the note here https://docs.sentry.io/platforms/
![image](https://github.com/getsentry/sentry-java/assets/4999776/95a903a0-3373-43dd-b1ee-93de3ea2d157)

However, since you're using your own instance of Hub, this is possible, but you will have to manually register available integrations and other things, pretty much what is done [here](https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/main/java/io/sentry/android/core/SentryAndroid.java#L92-L130) to achieve the same functionality. It's still not recommended though, as some of those integrations might clash with the Main Application in one or another way.
